### PR TITLE
Improve and develop trigger submission reporting

### PR DESCRIPTION
**Pull request by carlpulley-da**
When graphing or debugging trigger simulations, it helps to collect and collate the following information:
- for each batch of submissions, the submission completion outcome (i.e. success or failure type)
- the stats and percentile timings for each submission batch and their outcomes

### CODE REVIEW

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala [hunk 1]
```diff
@@ -54,26 +54,27 @@ abstract class TriggerMultiProcessSimulation extends AsyncWordSpec with Abstract
 
       Behaviors
         .supervise[Message] {
-          Behaviors.setup { context =>
+          Behaviors.setup { rootContext =>
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/TriggerMultiProcessSimulation.scala:57]: Changes here simplify understanding what Actor context is being used (and so what child actors are watched by the parent)

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/TriggerProcess.scala [hunk 2]
```diff
@@ -130,7 +130,7 @@ final class TriggerProcessFactory private[simulation] (
           (ref: ActorRef[LedgerRegistration.LedgerApi]) =>
             LedgerProcess.TriggerRegistration(
               LedgerRegistration
-                .Registration(triggerId, context.self, actAs, transactionFilter, ref)
+                .Registration(triggerId, triggerDefRef, context.self, actAs, transactionFilter, ref)
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/TriggerProcess.scala:133]: Allows reporting data to displayed alongside a more human friendly trigger name

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerApiClient.scala [hunk 3]
```diff
@@ -25,41 +30,109 @@ object LedgerApiClient {
...
+              Some(Status(cause.getStatus.getCode.value(), cause.getStatus.getDescription)),
+            )
+            replyTo ! TriggerProcess.MessageWrapper(TriggerMsg.Completion(completion))
+            report ! ReportingProcess.SubmissionUpdate(
+              SubmissionReporting.CompletionUpdate(
+                completionTimestamp,
+                commandId,
+                triggerId,
+                triggerDefRef,
+                completion,
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerApiClient.scala:91]: When command submissions failure (as part of the ledger request), we need to match real trigger behaviour and generate a "fake" completion failure signal.

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerRegistration.scala [hunk 4]
```diff
@@ -87,7 +88,17 @@ final class LedgerRegistration(client: LedgerClient)(implicit
           client.commandClient
             .completionSource(Seq(actAs.unwrap), offset)
             .collect { case CompletionElement(completion, _) =>
+              val timestamp = System.currentTimeMillis()
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/ledger/LedgerRegistration.scala:91]: Command submission completed

File: triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/report/MetricsReporting.scala [hunk 5]
```diff
@@ -91,4 +94,11 @@ private[simulation] object MetricsReporting {
         }
     }
   }
+
+  // We attempt to record just the completion summary prefix message (and not its detailed explanation)
+  private[this] def extractStatusMessage(status: GrpcStatus): String = {
+    val failureMessage = status.message.takeWhile('(' != _)
```
Comment (carlpulley-da) [inline triggers/tests/src/test/scala/com/digitalasset/daml/lf/engine/trigger/simulation/process/report/MetricsReporting.scala:100]: For reporting, we only want the status summary tag and not the whole status description

**Review by remyhaemmerle-da — APPROVED** [review]
Nice. Thanks.

**Pull request merged**